### PR TITLE
fix issues to get better performance and support small reduce factor

### PR DESCRIPTION
**Pull request by syang1993**
Hi,

I made some changes to prenet and the attention wrappers to support small reduce factor. Since in attention implementation in tensorflow:
https://github.com/tensorflow/tensorflow/blob/r1.8/tensorflow/contrib/seq2seq/python/ops/attention_wrapper.py#L1380, there is a default `_cell_input_fn` to concat the previous step's attention output and current input, which will affect the performance and alignment with small reduce factor. And also, I found using dropout in synthesis time will get better results than 'drop_rate=0'.

Here are some results using Nancy corpus after 100K training steps.
[samples.zip](https://github.com/keithito/tacotron/files/2129625/samples.zip)

### CODE REVIEW

File: models/tacotron.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import tensorflow as tf
-from tensorflow.contrib.rnn import GRUCell, MultiRNNCell, OutputProjectionWrapper, ResidualWrapper
+from tensorflow.contrib.rnn import GRUCell,LSTMCell, MultiRNNCell, OutputProjectionWrapper, ResidualWrapper
 from tensorflow.contrib.seq2seq import BasicDecoder, BahdanauAttention, AttentionWrapper
 from text.symbols import symbols
 from util.infolog import log
 from .helpers import TacoTestHelper, TacoTrainingHelper
 from .modules import encoder_cbhg, post_cbhg, prenet
-from .rnn_wrappers import DecoderPrenetWrapper, ConcatOutputAndAttentionWrapper
+from .rnn_wrappers import DecoderPrenetWrapper, ConcatOutputAndAttentionWrapper, ZoneoutWrapper
```
[the remark this answers is not in the document]
Comment (syang1993) [reply models/tacotron.py]: Hi, I did some experiments with zoneout, which is used in Tacotron-2. In this PR, I didn't use it, I'm sorry for this unused import.
